packages/agreement-process/src/services/agreementService.ts

```diff
@@ -621,14 +624,16 @@ export function agreementServiceBuilder(
         correlationId
       );
 
+      const canBeUpgraded = isSelfConsumer || (verifiedValid && declaredValid);
```

Comment:
@beetlecrunch can you confirm this is the correct behaviour?

Reply:
Yes, I confirm.